Source/Methods/Envelope Tracking/EnvelopeTracking.cs

```diff
@@ -0,0 +1,309 @@
+﻿using NationalInstruments.DataInfrastructure;
+using NationalInstruments.ModularInstruments;
+using NationalInstruments.ModularInstruments.NIRfsg;
+using NationalInstruments.ModularInstruments.SystemServices.TimingServices;
+using System;
+using System.Linq;
+
+namespace NationalInstruments.ReferenceDesignLibraries.Methods
+{
+    public static class EnvelopeTracking
+    {
+
+        #region Type Definitions
+        public struct EnvelopeGeneratorConfiguration
+        {
+            public RfsgTerminalConfiguration TerminalConfiguration;
+
+            public static EnvelopeGeneratorConfiguration GetDefault()
+            {
+                return new EnvelopeGeneratorConfiguration()
+                {
+                    TerminalConfiguration = RfsgTerminalConfiguration.Differential,
+                };
+            }
+        }

```

Comment:
Irrespective of my later comments on lines +209 to +214, why the separate struct from the TrackerConfiguration? I suppose you are probably trying to make it clear that these are instrument specific settings, while the items in the TrackerConfiguration really define attributes of the tracker that the generator needs to match? It just seems like they could go in the same struct perhaps.

Reply:
I'm pro removing terminal configuration since 5820 only supports differential. I say lets take out EnvelopeGeneratorConfiguration and that should resolve this.

Reply:
9561a5e4bd08e0f93528dfb8506b35d36dc549d7

Reply:
I have changed my mind on this, as it appears necessary for other reasons. See thread above on 207-208.

Reply:
resolving since no longer applicable. will continue on above thread.